test/e2e/storage/csi_mock_volume.go

```diff
@@ -383,7 +384,7 @@ var _ = utils.SIGDescribe("CSI mock volume", func() {
 			}.AsSelector().String()
 			msg := "AttachVolume.Attach failed for volume"
 
-			err = e2eevents.WaitTimeoutForEvent(m.cs, pod.Namespace, eventSelector, msg, framework.PodStartTimeout)
+			err = e2eevents.WaitTimeoutForEvent(m.cs, pod.Namespace, eventSelector, msg, f.Timeouts.PodStart)
```

Comment:
Hm so if a plugin decides to set a non-default timeout, then that will impact the timeout for non driver specific tests?

Is that what we want?

Reply:
can you point to an e.g for non-driver specific test?

Reply:
I was thinking specifically about this csi mock driver test. If you run the csi mock driver test along with a cloud plugin that takes minutes to provision/attach, etc, then that increases the timeout on the mock driver tests, which aren't expected to take very long, and could mask other problems in the system.

Reply:
I see, this seems to be a csi mock plugin specific problem. For the testsuites, they are run in the context of a driver (intree, csi or external). We can remove the non-default timeouts for the mock driver. thoughts?

Reply:
I guess right now these timeouts are only configurable via external storage interface or a custom ginkgo suite, so it's unlikely for someone to be running both csi mock and external storage tests in the same run. I would be more concerned if we want to extend this to all e2es in the future.

Reply:
Notice that this csi mock driver test is not using custom timeouts, it's using the default ones:

https://github.com/kubernetes/kubernetes/pull/96042/files/f594e0f7737a0d281e9b08ddf1b1ec770755ee6a#diff-ffda3f0c41f167e7deedad89b23cf9b89d0e9805f910c81e075baa19d667f21fR137

Even if it was using the custom timeouts, it would get the values returned by the mock driver's `GetTimeouts()` method, as in:

```diff
diff --git a/test/e2e/storage/drivers/csi.go b/test/e2e/storage/drivers/csi.go
index 03ec35e557c..d4956d47389 100644
--- a/test/e2e/storage/drivers/csi.go
+++ b/test/e2e/storage/drivers/csi.go
@@ -278,6 +278,7 @@ type CSIMockDriverOpts struct {
 var _ testsuites.TestDriver = &mockCSIDriver{}
 var _ testsuites.DynamicPVTestDriver = &mockCSIDriver{}
 var _ testsuites.SnapshottableTestDriver = &mockCSIDriver{}
+var _ testsuites.CustomTimeoutsTestDriver = &mockCSIDriver{}

 // InitMockCSIDriver returns a mockCSIDriver that implements TestDriver interface
 func InitMockCSIDriver(driverOpts CSIMockDriverOpts) testsuites.TestDriver {
@@ -336,6 +337,12 @@ func InitMockCSIDriver(driverOpts CSIMockDriverOpts) testsuites.TestDriver {
        }
 }

+func (m *mockCSIDriver) GetTimeouts() *framework.TimeoutContext {
+       timeouts := framework.NewTimeoutContextWithDefaults()
+       timeouts.PodStart = time.Second * 5
+       return timeouts
+}
+
 func (m *mockCSIDriver) GetDriverInfo() *testsuites.DriverInfo {
        return &m.driverInfo
 }
```

Reply:
Awesome, this addresses my concern for now thanks!